src/reducers/profile-view.js

```diff
@@ -228,10 +228,11 @@ function viewOptionsPerThread(
 
       return [
         ...state.slice(0, threadIndex),
-        Object.assign({}, state[threadIndex], {
```

Comment:
`Object.assign` doesn't keep the "readonly" state (at least with this version of Flow), but the object spread operator does. That's why I changed them all in this PR.